### ISSUE TITLE
This needs major updating to be usable

### ISSUE THREAD

**Issue by coryjreid** [pmusaraj/discourse-mobile-single-site-app#22]
It appears a lot has changed since this app was cobbled together. OAuth login flows are totally broken in the app. It can't even login with an email or password. And seeing all that will be a miracle because you have to fix things such as following the instructions [here](https://github.com/facebook/react-native/issues/26331) to even get the thing building properly.

The README should be updated to make this clear -or- the project updated.

Thanks, though!

**Comment by pmusaraj** [comment]
For most oauth providers, logins this should work, but you need to whitelist the correct URLs so the authentication happens in the webview. See https://github.com/pmusaraj/discourse-mobile-single-site-app/blob/master/default.variables.js#L24 

If you run into issues when building the app, please post the specific issues. Or, better yet, please contribute fixes (I built this repo 2.5 years ago and have not had to use it a lot recently).

**Comment by coryjreid** [comment]
Fair enough. I apologize for the frustrated message. I am new to iOS development and just wanted to quickly setup some forums and a corresponding single-site app with push notifications so all my users can be notified - and I wanted it to be simple and it's turning out not to be. I will see about pushing some fixes soon. :)

**Comment by renato** [comment]
@pmusaraj I'm not going to work on this so soon, but considering their current states do you think the best bet nowadays would be to fork this repo or [discourse/DiscourseMobile](/discourse/DiscourseMobile) itself as baseline for a single site app?

Off-topic/Curiosity: I didn't know this repo existed and was set to work on DiscourseMobile, but [this topic](https://meta.discourse.org/t/convert-forum-to-ios-app/86885/3) appeared on meta's /latest ~30min ago, but there's no recent interaction there.

**Comment by pmusaraj** [comment]
@renato it so happens that I am updating this repo. Might take another few weeks, but it is not dead!

**Comment by afdy** [comment]
How you getting in with the update? I’m intrigued (and super keen to see if push notifications will be fixed!).

**Comment by pmusaraj** [comment]
@renato @afdy I have just merged a major update to the repo. I think this is now in very good shape, please give it a try. Make sure you try it on a fresh clone and using the updated `discourse-onesignal` plugin. 

The repo now includes Fastlane scripts that simplify greatly the task of adapting this whitelisted repo for your own app (and its assets, certificates, naming and configuration). Make sure you review the updated readme file and let me know if there are any mistakes in there.

**Comment by afdy** [comment]
Awesome thanks - just had a blast, couple of points so far;
- new guide doesn't mention creating app.variables.js 
- fastlane swith fails with "[07:36:58]: No such file or directory @ rb_sysopen - EastleighOnline/android-release.keystore" as I wasn't releasing on android.  I guess i have no excuse but to make that work now! ;)

I'll continue later!

**Comment by pmusaraj** [comment]
> * new guide doesn't mention creating app.variables.js

Right, because it's one of the items in the sampe folder. Maybe I'll make it more explicit. 

> * as I wasn't releasing on android

Ah right, you can probably comment out the android lines in that fastlane task in `Fastfile`. 

Thanks for trying this out.

**Comment by afdy** [comment]
I figured I'd go full steam ahead and give the android compile a a whirl, more feedback:
- Had to upgrade the OneSignal library so it didn't support the old UIWebView API (I went from 9.3.2 to 3.9.3).
- Bumped into next error which I've not yet figured out on fastlane android release - 

> > Task :app:processReleaseResources FAILED
> > Task :app:bundleReleaseResources FAILED
> 
> FAILURE: Build completed with 2 failures.
> 
> 1: Task failed with an exception.
> -----------
> * What went wrong:
> Execution failed for task ':app:processReleaseResources'.
> > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
>    > Android resource linking failed
>      AAPT: error: resource style/SplashScreen_SplashTheme (aka com.eastleigh.online.app:style/SplashScreen_SplashTheme) not found.
>      error: failed linking references.
>      
>

**Comment by afdy** [comment]
> Task :app:processReleaseResources FAILED
> Task :app:bundleReleaseResources FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:processReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     AAPT: error: resource style/SplashScreen_SplashTheme (aka com.myapp.app:style/SplashScreen_SplashTheme) not found.
     error: failed linking references.

**Comment by afdy** [comment]
I found another issue;

- I had to upgrade the OneSignal library to not support UIWebView (against apple guidelines).  I upgraded to 3.9.3 to make it work, but there are later versions.
- I went to attempt android build so generated a keystore (which fixed iOS builds), but then bumped into another error:

I am struggling with where to start here, some brief googling around suggests its due to google library version issues.

> Task :app:processReleaseResources FAILED
> Task :app:bundleReleaseResources FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:processReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     AAPT: error: resource style/SplashScreen_SplashTheme (aka com.myapp.app:style/SplashScreen_SplashTheme) not found.
     error: failed linking references.

**Comment by afdy** [comment]
Retried today from scratch, in case it was the OneSignal upgrade that broke it - but no, same error.

**Comment by afdy** [comment]
Got it - I had to add react-native-splash-screen and then the android version built too.  Missing from package.json.  That was a hard one to track.

**Comment by pmusaraj** [comment]
Thanks for trying this out and for the feedback. I have made some updates in the `main` branch: 

- updated OneSignal to 3.93, that's a good idea
- removed the splashscreen stuff for Android, it wasn't hooked up properly, and in the vast majority of the devices, the app loads fast enough not to need that screen
- removed another dependency that was no longer being used

**Comment by afdy** [comment]
Cool thanks - just tried a fresh run and it fails with the same error as previous until I add react-native-splash-screen. Fastlane build process must call it somewhere.

**Comment by pmusaraj** [comment]
Hmm... it builds on Android for me without that package. It's possible you have some leftover references from the previous `switch` command in the Android files.

**Comment by afdy** [comment]
Hmm I'd say not possible, I checked out a whole new directory, only files that persist are ones I've copied over in .env.myapp, and the contents of the myapp folder.

I've just recreated again and it does exactly the same thing.

`git log` shows I definitely pulled in your commits. :)

I see SplashScreen_SplashTheme is still defined here:
https://github.com/pmusaraj/discourse-mobile-single-site-app/blob/80eb9b47636548df58d0745c487fde6e48d3f7ec/android/app/src/main/res/values/styles.xml

**Comment by afdy** [comment]
Removed this bit:

```
    <style name="SplashScreenTheme" parent="SplashScreen_SplashTheme">
        <item name="colorPrimaryDark">@color/splashprimary</item>
    </style>
```

from styles.xml and it builds without it now.

**Comment by pmusaraj** [comment]
Indeed, I had left those lines in during the last fix. It's fixed now though.

**Comment by afdy** [comment]
Awesome, this now works great thanks!  Awesome work by the way, really useful stuff!

Love fastlane - is there any way fastlane can take over the build numbering in the custom env files? Right now things get up if I pull a clean version again as the build numbers go backwards! :)

**Comment by afdy** [comment]
Love that there is no longer a need to open XCode - how hard would it be to do an iPad optimised version?

**Comment by pmusaraj** [comment]
Yes, those are two good and fairly doable improvements: envfiles controlling the build version and number, and support for iPad. Not sure when I'll get to the them, tbh. PRs are certainly welcome!
